Mobile Buy SDK/Mobile Buy SDK/Operations/BUYRequestOperation.m

```diff
@@ -169,14 +168,24 @@ - (NSURLSessionDataTask *)request:(NSURLRequest *)request completion:(BUYRequest
 			json = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
 		}
 		
-		NSHTTPURLResponse *httpResponse = (id)response;
-		if (!httpResponse.successful && !error) {
-			error = [[NSError alloc] initWithDomain:kShopifyError code:httpResponse.statusCode userInfo:json];
+		NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+		if (!error) {
+			error = [self errorFromJSON:json response:httpResponse];
 		}
 		
 		completion(json, httpResponse, error);
 	}];
 	return task;
 }
 
+#pragma mark - Error -
+
+- (NSError *)errorFromJSON:(NSDictionary *)json response:(NSHTTPURLResponse *)response
```

Comment:
Very true. Somehow missed that I created a category on `NSHTTPURLResponse` to provide a `success` flag based on whether `the statusCode` is a `2xx` or not. Using that within the caller eliminates the need to this method altogether. Will remove.